packages/widget/src/context/config.ts

```diff
@@ -0,0 +1,72 @@
+import { customElement, property } from 'lit/decorators.js';
+import { createContext, provide } from '@lit/context';
+import type { WalletConnectOptions } from '@web3-onboard/walletconnect/dist/types';
+import type { HTMLTemplateResult } from 'lit';
+import { html } from 'lit';
+import type { AppMetadata } from '@web3-onboard/common';
+import { BaseComponent } from '../components/common/base-component';
+import type { Theme } from '../interfaces';
+
+export interface ConfigContext {
+  theme?: Theme;
+  walletConnectOptions?: WalletConnectOptions;
+  appMetaData?: AppMetadata;
+}
+
+export const configContext = createContext<ConfigContext>(
+  Symbol('sygma-config-context')
+);
+
+declare global {
+  interface HTMLElementEventMap {
+    configUpdate: ConfigUpdateEvent;
+  }
+}
+
+export class ConfigUpdateEvent extends CustomEvent<Partial<ConfigContext>> {
```

Comment:
Does it make sense for child component to update config?

Reply:
I think not, I was thinking about theming here mostly, but those properties are static.